evm_arithmetization/src/cpu/kernel/aggregator.rs

```diff
@@ -7,167 +7,173 @@ use super::assembler::{assemble, Kernel};
-        include_str!("asm/memory/packing.asm"),
-        include_str!("asm/memory/syscalls.asm"),
-        include_str!("asm/memory/txn_fields.asm"),
-        include_str!("asm/mpt/accounts.asm"),
-        include_str!("asm/mpt/delete/delete.asm"),
-        include_str!("asm/mpt/delete/delete_branch.asm"),
-        include_str!("asm/mpt/delete/delete_extension.asm"),
-        include_str!("asm/mpt/hash/hash.asm"),
-        include_str!("asm/mpt/hash/hash_trie_specific.asm"),
-        include_str!("asm/mpt/hex_prefix.asm"),
-        include_str!("asm/mpt/insert/insert.asm"),
-        include_str!("asm/mpt/insert/insert_extension.asm"),
-        include_str!("asm/mpt/insert/insert_leaf.asm"),
-        include_str!("asm/mpt/insert/insert_trie_specific.asm"),
-        include_str!("asm/mpt/read.asm"),
-        include_str!("asm/mpt/storage/storage_read.asm"),
-        include_str!("asm/mpt/storage/storage_write.asm"),
-        include_str!("asm/mpt/util.asm"),
-        include_str!("asm/rlp/decode.asm"),
-        include_str!("asm/rlp/encode.asm"),
-        include_str!("asm/rlp/encode_rlp_scalar.asm"),
-        include_str!("asm/rlp/encode_rlp_string.asm"),
-        include_str!("asm/rlp/increment_bounded_rlp.asm"),
-        include_str!("asm/rlp/num_bytes.asm"),
-        include_str!("asm/rlp/read_to_memory.asm"),
-        include_str!("asm/shift.asm"),
-        include_str!("asm/signed.asm"),
-        include_str!("asm/journal/journal.asm"),
-        include_str!("asm/journal/account_loaded.asm"),
-        include_str!("asm/journal/account_destroyed.asm"),
-        include_str!("asm/journal/account_touched.asm"),
-        include_str!("asm/journal/balance_transfer.asm"),
-        include_str!("asm/journal/nonce_change.asm"),
-        include_str!("asm/journal/storage_change.asm"),
-        include_str!("asm/journal/storage_loaded.asm"),
-        include_str!("asm/journal/code_change.asm"),
-        include_str!("asm/journal/refund.asm"),
-        include_str!("asm/journal/account_created.asm"),
-        include_str!("asm/journal/revert.asm"),
-        include_str!("asm/journal/log.asm"),
-        include_str!("asm/transactions/common_decoding.asm"),
-        include_str!("asm/transactions/router.asm"),
-        include_str!("asm/transactions/type_0.asm"),
-        include_str!("asm/transactions/type_1.asm"),
-        include_str!("asm/transactions/type_2.asm"),
-        include_str!("asm/transactions/type_3.asm"),
-        include_str!("asm/util/assertions.asm"),
-        include_str!("asm/util/basic_macros.asm"),
-        include_str!("asm/util/keccak.asm"),
-        include_str!("asm/util/math.asm"),
-        include_str!("asm/account_code.asm"),
-        include_str!("asm/balance.asm"),
-        include_str!("asm/bloom_filter.asm"),
-    ];
+pub static KERNEL: Lazy<Kernel> = Lazy::new(combined_kernel);
 
+pub(crate) fn combined_kernel_from_files(files: &[&str]) -> Kernel {
     let parsed_files = files.iter().map(|f| parse(f)).collect_vec();
```

Comment:
(I wonder if this could be `const` some day)

Reply:
I guess you can. The only "problem" would be that one of the tests uses a `Vec` of files for the kernel.

Reply:
Nah, it's not possible today. But it should be!